kv/dist_sender_test.go

```diff
@@ -752,3 +752,99 @@ func TestGetNodeDescriptor(t *testing.T) {
 		return util.Errorf("wanted NodeID 5, got %v", desc)
 	})
 }
+
+// TestSpanRangeScanWhenMerging simulate a range merge operation happend between get range desc
+// and get next range desc in getDescriptors function. the second sub scan use next range's start key
+// as start key of scan, this cause the the whole scan get duplicate key-values.
+func TestSpanRangeScanWhenMerging(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	g, s := makeTestGossip(t)
+	defer s()
+
+	// Asume we have two range [a-b) [b-KeyMax)
+	merged := false
+
+	var firstRange = proto.RangeDescriptor{
+		RaftID:   1,
+		StartKey: proto.Key("a"),
+		EndKey:   proto.Key("b"),
+		Replicas: []proto.Replica{
+			{
+				NodeID:  1,
+				StoreID: 1,
+			},
+		},
+	}
+
+	// Two ranges will be merged to one range
```

Comment:
The merged descriptor, which will be looked up after having processed the stale range `[a,b)`.